mysql-test/main/auto_increment_ranges_innodb.test

```diff
@@ -18,3 +18,59 @@ select * from t1;
 drop table t1;
 --let $datadir=`select @@datadir`
 --remove_file $datadir/test/load.data
+
+--echo #
+--echo # MDEV-21842: auto_increment does not increment with compound primary
+--echo # key on partitioned table
+--echo #
+
+create or replace table `t` (
+  `id` bigint(20) unsigned not null auto_increment,
+  `a` int(10) not null ,
+  `dt` date not null,
+  primary key (`id`, `dt`) ,
+  unique key (`a`, `dt`)
+)
+ partition by range  columns(`dt`)
+(
+ partition `p202002` values less than ('2020-03-01'),
+ partition `P202003` values less than ('2020-04-01')
+);
+
+connect (con1, localhost, root,,);
+connect (con2, localhost, root,,);
+
+--connection con1
+start transaction;
+insert into t (a, dt) values (1, '2020-02-29');
+
+--connection con2
+start transaction;
+send insert into t (a, dt) values (1, '2020-02-29');
+
+# Sleep a second to ensure that the above insert via conn2 reaches the server
+# and increment next_auto_inc_val before the following insert via con1 reaches
+# the server. The ordering is fundamental for this test case.
+--sleep 1
```

Comment:
The `send` command seems to execute the given query asynchronously. 
https://travis-ci.org/github/MariaDB/server/jobs/740501831

Reply:
Please never use `sleep` for solving race conditions: it never guarantees the right order of execution; it slows down tests.

Reply:
Otherwise your patch looks good.

Reply:
Thank you for your review, midenok.

I agree that the use of the sleep command should be avoided. I will try to fix it.

Reply:
@nayuta-yanagisawa the correct way is to use debug_sync

Reply:
Thanks, I will try it!

Reply:
I finally decided to use the [wait-condition](https://dev.mysql.com/doc/internals/en/wait-condition.html), because it seems to fit my use case. Thank you for your advice, which helps me to find the wait-condition technique from the MySQL internal manual.